src/main/java/net/snowflake/client/jdbc/SnowflakeBaseResultSet.java

```diff
@@ -1366,12 +1368,17 @@ public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
         Object object = getObject(columnIndex);
         if (object == null) {
           return null;
-        } else if (object instanceof JsonSqlInput) {
-          JsonNode jsonNode = ((JsonSqlInput) object).getInput();
-          return (T)
-              OBJECT_MAPPER.convertValue(jsonNode, new TypeReference<Map<String, Object>>() {});
+        } else if (object instanceof Map) {
+          throw new SQLException(
+              "Arrow native struct couldn't be converted to String. To map to SqlData the method getObject(int columnIndex, Class type) should be used");
         } else {
-          return (T) ((ArrowSqlInput) object).getInput();
+          try {
+            return (T)
+                OBJECT_MAPPER.readValue(
+                    (String) object, new TypeReference<Map<String, Object>>() {});
```

Comment:
Map can have integer/long keys, will it work?

Reply:
Changed to Object as map key